assets/js/googlesitekit/notifications/datastore/notifications.js

```diff
@@ -418,9 +440,46 @@ export const selectors = {
 	 */
 	isNotificationDismissed: createRegistrySelector(
 		( select ) => ( state, id ) => {
+			const notification =
+				select( CORE_NOTIFICATIONS ).getNotification( id );
+
+			if ( notification.dismissRetries > 0 ) {
+				return select( CORE_USER ).isPromptDismissed( id );
+			}
+
 			return select( CORE_USER ).isItemDismissed( id );
 		}
 	),
+	/**
+	 * Determines whether a notification that can reappear again for a fixed number of times
+	 * on dismissal is at its final appearance.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @param {string} id    Notification id.
+	 * @return {(boolean|undefined)} TRUE if notification is on its final retry, otherwise FALSE, `undefined` if not resolved yet.
+	 */
+	isNotificationRetryFinal: createRegistrySelector(
+		( select ) => ( state, id ) => {
+			const notification =
+				select( CORE_NOTIFICATIONS ).getNotification( id );
+
+			// If a notification does not have retries, it always will be on its final render.
+			if ( notification.dismissRetries === 0 ) {
+				return true;
+			}
```

Comment:
What do you think about considering `dismissible` here as well? What if the notification isn't dismissible? Maybe it should trigger an invariant violation? I feel like true or false would both be incorrect 🤔

Also, maybe `Retry` should be `Dismiss`?

Reply:
Originally, the idea here was for the selector to be used in the common `<Dismiss>` component that would've been used by all notifications. Added an invariant now like you suggest.